elephant/statistics.py

```diff
@@ -974,23 +977,414 @@ def complexity_pdf(spiketrains, bin_size):
            pp. 96-114, Springer Berlin Heidelberg, 2008.
 
     """
-    # Computing the population histogram with parameter binary=True to clip the
-    # spike trains before summing
-    pophist = time_histogram(spiketrains, bin_size, binary=True)
-
-    # Computing the histogram of the entries of pophist (=Complexity histogram)
-    complexity_hist = np.histogram(
-        pophist.magnitude, bins=range(0, len(spiketrains) + 2))[0]
-
-    # Normalization of the Complexity Histogram to 1 (probabilty distribution)
-    complexity_hist = complexity_hist / complexity_hist.sum()
-    # Convert the Complexity pdf to an neo.AnalogSignal
-    complexity_distribution = neo.AnalogSignal(
-        np.expand_dims(complexity_hist, axis=1) *
-        pq.dimensionless, t_start=0 * pq.dimensionless,
-        sampling_period=1 * pq.dimensionless)
-
-    return complexity_distribution
+    warnings.warn("complexity_pdf is deprecated in favor of the complexity "
+                  "class which has a pdf attribute. complexity_pdf will be "
+                  "removed in the next Elephant release.", DeprecationWarning)
+
+    complexity = Complexity(spiketrains, bin_size=bin_size)
+
+    return complexity.pdf()
+
+
+class Complexity(object):
+    """
+    Class for complexity distribution (i.e. number of synchronous spikes found)
+    of a list of `neo.SpikeTrain` objects.
+
+    Complexity is calculated by counting the number of spikes (i.e. non-empty
+    bins) that occur separated by `spread - 1` or less empty bins, within and
+    across spike trains in the `spiketrains` list.
+
+    Implementation (without spread) is based on [1]_.
+
+    Parameters
+    ----------
+    spiketrains : list of neo.SpikeTrain
+        Spike trains with a common time axis (same `t_start` and `t_stop`)
+    sampling_rate : pq.Quantity or None, optional
```

Comment:
Missing description of behavior when None

Reply:
Done